native/packages/react-native-bpk-component-picker/src/BpkPickerMenu.android.js

```diff
@@ -27,64 +33,109 @@ import {
   lineHeightBase,
   spacingBase,
 } from 'bpk-tokens/tokens/base.react.native';
+import BpkPickerItem from './BpkPickerItem';
 
-const rowsToDisplay = 6;
-
-const maxListHeight =
-  // eslint-disable-next-line no-mixed-operators
-  (2 * spacingBase + lineHeightBase) * rowsToDisplay;
+const MAX_ROWS_TO_DISPLAY = 6;
 
 const styles = StyleSheet.create({
   overlay: {
-    flex: 1,
+    left: 0,
+    height: '100%',
+    position: 'absolute',
+    top: 0,
+    width: '100%',
     backgroundColor: setOpacity(colorGray900, 0.8),
-    justifyContent: 'center',
   },
   list: {
     backgroundColor: colorWhite,
     borderRadius: borderRadiusSm,
-    maxHeight: maxListHeight,
-    margin: spacingBase,
-    flex: 0,
+    alignSelf: 'center',
+    top: '50%',
+    minWidth: '50%',
   },
 });
 
-const PickerMenu = props => {
+const BpkPickerMenu = props => {
   const { visible, children, onValueChange, onClose, selectedValue } = props;
-  const pickerItems = React.Children.map(children, (child, index) =>
-    React.cloneElement(child, {
-      selected: selectedValue && selectedValue === child.props.value,
-      onPress: value => {
-        onValueChange(value, index);
-        onClose();
-      },
-    }),
-  );
+
+  let initialScrollIndex = 0;
+  const pickerItems = React.Children.map(children, (child, index) => {
+    const { label, value } = child.props;
+
+    let selected = false;
+    if (selectedValue && selectedValue === value) {
+      selected = true;
+
+      // If selected item will be off-screen, set initialScrollIndex.
+      if (index >= MAX_ROWS_TO_DISPLAY) {
+        initialScrollIndex = index - 2;
```

Comment:
is the -2 to have the selected item centered?

Reply:
Yeah, I'll add a screenshot to this thread, one moment 🕙

Reply:
Do you think I should add a comment to the code to make it clearer?

Reply:
Without `- 2` (selected item is at the top)
![screenshot_1521627144](https://user-images.githubusercontent.com/73652/37704326-69aca3c6-2cf0-11e8-8d99-d8bca95bbcae.png)

With `- 2`: (selected item is centred nicely)
![screenshot_1521627160](https://user-images.githubusercontent.com/73652/37704344-763a9918-2cf0-11e8-8069-c921655b48d2.png)

Reply:
Maybe a tiny comment can help out why the magic 2 but I gueesed it right first time so should be ok if you don't add it 😄

Reply:
I added it anyway to be safe.